service/src/main/java/com/ge/predix/acs/service/policy/validation/PolicySetValidatorImpl.java

```diff
@@ -102,14 +166,35 @@ private void validatePolicyActions(final Policy p) {
             }
             for (String action : policyActions.split("\\s*,\\s*")) {
                 if (!this.validAcsPolicyHttpActionsSet.contains(action)) {
-                    throw new PolicySetValidationException(String.format("Policy Action validation failed: "
-                                    + "the action: [%s] is not contained in the allowed set of actions: [%s]", action,
-                            this.validAcsPolicyHttpActions));
+                    throw new PolicySetValidationException(String.format(
+                            "Policy Action validation failed: "
+                                    + "the action: [%s] is not contained in the allowed set of actions: [%s]",
+                            action, this.validAcsPolicyHttpActions));
                 }
             }
         }
     }
 
+    private void vaLidatePolicyObligations(final List<String> obligationIds, final Policy policy) {
+        List<String> notFound = new ArrayList<>();
+        for (String obligationId : policy.getObligationIds()) {
+            if (StringUtils.isEmpty(obligationId)) {
```

Comment:
Should unload to schema validation if possible

Reply:
I don't think this is  possible, for list<string>obligationids, it can be empty or populated and the only validation is that there are not repeated or blank or empty string values, don't think javax validation annotations allows you to specify that the contents of a collection should not be null or blank, the annotations are applied to the collection itself